### PR TITLE
Remove old streaming reference doc

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

Follow-up to #10709

Now that the [new streaming reference](/docs/metrics-storage-management/refrence-streaming.mdx) is published, we can remove almost all of the old content. I left the file in place with a little bit of information about the new user documentation in case there's a need to create _developer_ documentation in `/streaming/README.md`, which wouldn't be published on [Netdata Learn](https://learn.netdata.cloud).

> For some context on this and similar PRs, I'm slowly moving all user documentation to the `/docs` folder. This should improve how we organize and publish the docs on [Netdata Learn](https://learn.netdata.cloud), while also leaving the option to create developer documentation in various `README.md` files scattered across the codebase. By separating user vs. developer docs, we should improve the experience for both.

##### Component Name

area/streaming

### CODE REVIEW

File: daemon/config/README.md [hunk 1]
```diff
@@ -131,7 +132,8 @@ To understand what this section is and how it should be configured, please refer
 
 ### [backend]
 
-Refer to the [streaming and replication](/streaming/README.md) documentation.
+The backends system is deprecated. Refer to [exporting](/exporting/README.md) for details on sending metrics to external
```
Comment (thiagoftsm) [inline daemon/config/README.md:135]: Hello @joelhans , the system name is `backend`, why did you write a plural here?

Comment (joelhans) [reply daemon/config/README.md]: Good question! I'm not sure why I did, but it's removed now.

**Review by thiagoftsm — APPROVED** [review]

**Comment by ilyam8** [comment]
Is this still our plan - removing  [streaming/README.md](https://github.com/netdata/netdata/tree/master/streaming#streaming-and-replication)? @stelfrag @odynik

**Review by DShreve2 — APPROVED** [review]

**Comment by ilyam8** [comment]
Closing because there are merge conflicts. cc @tkatsoulas

**Pull request closed without merging**